mne/decoding/tests/test_time_gen.py

```diff
@@ -252,24 +287,26 @@ def test_generalization_across_time():
     # and http://bit.ly/1u7t8UT
     assert_raises(ValueError, gat.score, epochs2)
     gat.score(epochs)
-    scores = sum(scores, [])  # flatten
     assert_true(0.0 <= np.min(scores) <= 1.0)
     assert_true(0.0 <= np.max(scores) <= 1.0)
 
     # Test that gets error if train on one dataset, test on another, and don't
     # specify appropriate cv:
-    gat = GeneralizationAcrossTime()
+    if check_version('sklearn', '0.18'):
+        cv = ShuffleSplit()
```

Comment:
can you do `cv = _check_cv(method, n_iter)` and then in `check_cv` do the bulk of the `if-else` to avoid branching multiple times? Does it make the code cleaner?